### PR TITLE
WIP - GRANT-330 & GRANT-331 - populate detail fields

### PR DESCRIPTION
**Pull request by coryatwater-FW**
WIP

### CODE REVIEW

File: api/src/application/application.service.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class ApplicationService {
 
   async getApplication(applicationId: number): Promise<Application> {
     const application = await this.applicationRepository.findOne(applicationId, {
-      relations: ['assignedTo', 'form', 'lastUpdatedBy'],
+      relations: ['assignedTo', 'form'],
```
Comment (alva-fresh) [inline api/src/application/application.service.ts:72]: We don't need `lastUpdatedBy` in this case?

Comment (alva-fresh) [reply api/src/application/application.service.ts:72]: I think it's still used in `assignToUser`:
```
const application = await this.getApplication(applicationId);
    const user = await this.userService.getUser(assignToUserDto.userId);
    application.assignedTo = user;
    application.lastUpdatedByUserId = loggedInUser.userName;
    await this.applicationRepository.save(application);
```

File: client/components/application/RenderCHFSElement.tsx [hunk 2]
```diff
@@ -1,117 +1,9 @@
...
-  'select',
-  'currency',
-  'day',
-  'phoneNumber',
-];
+import { NOT_TO_BE_RENDERED, renderElement } from 'helpers';
 
 const NESTED_COMPONENTS = ['simplecols2'];
 
-const NOT_TO_BE_RENDERED = ['button', 'simplecontent', 'htmlelement'];
```
Comment (alva-fresh) [inline client/components/application/RenderCHFSElement.tsx:21]: Good thinking on moving it to `helpers`! Looks a lot cleaner

**Comment by alva-fresh** [comment]
Looks like fields with keys like radios and checkboxes display the key instead of label as a value.
For example, here:
Screenshot 2023-08-15 at 7 00 32 PM
should be "Regional district" I feel like

**Comment by alva-fresh** [comment]
_Phone_ and _email_ missing in Section 1

File: api/src/application/application.service.ts (hunk 1 above)
Comment (alva-fresh) [reply api/src/application/application.service.ts:72]: I feel like we need to capture who it was updatedBy in application

**Comment by alva-fresh** [comment]
> Looks like fields with keys like radios and checkboxes display the key instead of label as a value. For example, here: Screenshot 2023-08-15 at 7 00 32 PM should be "Regional district" I feel like

```
const organizeFieldsetData = (e: any, formData: any, componentKey?: any) => {
  return e?.components
    ?.filter((i: any) => !NOT_TO_BE_RENDERED.includes(i.type))
    .map((c: any) => {
      if (Array.isArray(c.rows) && c.rows.length > 0) {
        return c.rows?.map((r: any) =>
          renderGeneralField(r[1].components[0], formData, componentKey),
        );
      } else {
        switch (c.type) {
          case 'columns':
            return c.columns?.map((eachCol: any) =>
              eachCol?.components?.map((ec: any) => renderGeneralField(ec, formData, componentKey)),
            );

          case 'simpleradios':
          case 'simpleradioadvanced':
            return renderRadioValue(c, formData, componentKey);

          default:
            return renderGeneralField(c, formData, componentKey);
        }
      }
    });
};

const renderRadioValue = (e: any, data: any, container?: any) => {
  const label = e.properties?.portalWording ?? e.label;
  const value = data[container][e.key];

  return (
    <div key={e.id} className='w-fit grid grid-flow-row'>
      <span className='font-bold'>{label}</span>
      <span key={e.key}>{`${
        e.values?.find((item: any) => item.value === value)?.label ?? '-'
      }`}</span>
    </div>
  );
};
```

**Comment by alva-fresh** [comment]
> _Phone_ and _email_ missing in Section 1

```
switch (c.type) {
          case 'columns':
          case 'simplecols2':
            return c.columns?.map((eachCol: any) =>
              eachCol?.components?.map((ec: any) => renderGeneralField(ec, formData, componentKey)),
            );

          case 'simpleradios':
          case 'simpleradioadvanced':
            return renderRadioValue(c, formData, componentKey);

          default:
            return renderGeneralField(c, formData, componentKey);
        }
```

**Comment by alva-fresh** [comment]
Let's change the branch to summary table. I'm sorry, I think it's best to keep `cleanup` branch in review for now.
I made a PR only for changing where this PR points to, as well fixing some of the rendering values. Please take a look here: https://github.com/bcgov/BCAT/pull/19

**Comment by coryatwater-FW** [comment]
> Let's change the branch to summary table. I'm sorry, I think it's best to keep `cleanup` branch in review for now. I made a PR only for changing where this PR points to, as well fixing some of the rendering values. Please take a look here: #19

ok, I'll close this

**Pull request closed without merging**